pyccel/codegen/printing/ccode.py

```diff
@@ -1349,7 +1349,7 @@ def _print_For(self, expr):
             step  = self._print(expr.iterable.step )
         else:
             raise NotImplementedError("Only iterable currently supported is Range")
-        return ('for ({target} = {start}; {target} < {stop}; {target} += '
+        return ('for ({target} = {start}; (({step} > 0 ? 1 : -1)*({target})) < (({step} > 0 ? 1 : -1)*({stop})); {target} += '
```

Comment:
you can break this statement into 2 case:
- case 1 : target is known (Literal)
     in this case we can  know it target is positive or negative, so additional check are not needed in the generated code
     like you were doing in a4258ad
- case 2 : target is unknown (Variable)
     in this case you can keep what you did here
     you can improve printed code by using IfTernaryOperator  see ast/operators 
     ```
     {step} > 0 ? 1 : -1
     ```
     is equivalent to ```IfTernaryOperator(PyccelGt(step, 0), 1, -1)```
     or more better ```IfTernaryOperator(PyccelGt(step, 0), target, PyccelUnarySub(target))```
     otherwise you can keep what you did.